### PR TITLE
l2tpv3: T2352: check that the local-ip exists

### PR DESCRIPTION
**Pull request by thomas-mangin**

### CODE REVIEW

**Review by dmbaturin — APPROVED** [review]

**Comment by c-po** [comment]
Why not make use of an already existing function? `vyos.validate.is_addr_assigned()` ?

**Review by c-po — CHANGES_REQUESTED** [review]
Why not make use of an already existing function? `vyos.validate.is_addr_assigned()` ? The wheel should not be invented multiple times.

**Comment by thomas-mangin** [comment]
Agree this is duplication. I thought that validation was performing data validation (regex / checks) this is a query of the operation of the router, hence why I placed it under operational but I think you are right. opening a task to discuss.

**Comment by c-po** [comment]
No Task needed - please use it. The file has been called validate b/c it is called by most verify() functions which is true here, too.

**Comment by thomas-mangin** [comment]
kept the function as a separate commit. using the validate function in the code.

**Comment by c-po** [comment]
This means we have unused code now? Not a fan of it.

**Comment by thomas-mangin** [comment]
Just merge half the patch.

File: src/conf_mode/interfaces-l2tpv3.py [hunk 1]
```diff
@@ -170,6 +175,9 @@ def verify(l2tpv3):
     if not l2tpv3['local_address']:
         raise ConfigError(f'Must configure the l2tpv3 local-ip for {interface}')
 
+    if not l2tpv3['assigned']:
```
Comment (c-po) [inline src/conf_mode/interfaces-l2tpv3.py]: Why not call `is_addr_assigned(l2tpv3['local_address'])` here directly?

File: src/conf_mode/interfaces-l2tpv3.py [hunk 2]
```diff
@@ -198,7 +206,7 @@ def apply(l2tpv3):
     conf = deepcopy(L2TPv3If.get_config())
 
     # Check if L2TPv3 interface already exists
-    if l2tpv3['intf'] in interfaces():
+    if l2tpv3['intf'] in l2tpv3['interfaces']:
```
Comment (c-po) [inline src/conf_mode/interfaces-l2tpv3.py]: Why use a cached version of `interfaces()``

**Review by c-po — CHANGES_REQUESTED** [review]

**Pull request merged**